compiler/src/yul/runtime/functions/structs.rs

```diff
@@ -0,0 +1,133 @@
+use crate::yul::names;
+use fe_analyzer::namespace::types::Struct;
+use yultsur::*;
+
+/// Generate a YUL function that can be used to create an instance of
+/// `struct_type`
+pub fn generate_new_fn(struct_type: &Struct) -> yul::Statement {
+    let function_name = names::struct_new_call(&struct_type.name);
+
+    if struct_type.is_empty() {
+        let body = statement! { return_val := alloc(0) };
+        return function_definition! {
+            function [function_name]() -> return_val {
+                 [body]
```

Comment:
Could we just return `0` here?

Reply:
But what we return here is a pointer address if I return `0` doesn't this end up as being wrongfully interpreted as the pointer address where the (non-existent) struct is being stored. As far as I understand the pointer address `0` actually stores the highest available pointer so we would end up pointing there, no?

Reply:
> As far as I understand the pointer address 0 actually stores the highest available pointer so we would end up pointing there, no?

That's correct, but if it's an empty struct, we should be able to make the assumption that we'll never write or read to the address. Using `alloc` here performs a read and write on the `0x00` address that has no effect , since we're allocating `0` bytes. If anything, we would use the `avail` function to get the highest available memory pointer (which would return the same thing that `alloc(0)` does). But again, since this pointer is never used, we could just simplify things and return `0`. If there's a situation where we write to an empty struct, then we have a bug.

Reply:
Alright, that makes sense. I'll do the change and will put a comment on it to explain it.